### DISCUSSION TITLE
Cognito redirecting to localhost on production amplify

### DISCUSSION THREAD

**Discussion by wallacerenan** [nextauthjs/next-auth#2771] [Help]
### Question 💬

My application is not redirecting to a correct url when going up in amplify,

It's pointing to localhost: 3000

My environment variable is correct:
NEXTAUTH_URL = https://main.*******.amplifyapp.com

And my callback url is set up correctly in AWS Cognito:
https://main.*******.amplifyapp.com/api/auth/callback/cognito

I'm using the next 11

And actual version of "next-auth": "^3.29.0"

On login the next redirects to localhost:3000/api/auth/signin/cognito

### How to reproduce ☕️

`
import { signIn, signOut, useSession } from 'next-auth/client'

export default function HomePage() {
  const [session, loading] = useSession()

  if (loading) return <div>Carregando....</div>

  return (
    <>
      {!session && (
        <>
          Não logado <br />
          <button
            onClick={() =>
              signIn()
            }
          >
            Entrar
          </button>
        </>
      )}
      {session && (
        <>
          Logado como {session.user.email} <br />
          <button onClick={() => signOut()}>Sair</button>
        </>
      )}
    </>
  )
}`

![giphy](https://user-images.githubusercontent.com/79864236/133637682-df92194f-b6b9-47eb-ad55-95d2135bfcb3.gif)

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
Unfortunately this is most certainly a misconfiguration of the NEXTAUTH_URL environment variable. Try to double check by printing it out for example. We only default to localhost, if we didn't find the `process.env.NEXTAUTH_URL` value to be set.

**Reply by andreas-groos** [reply]
Same issue here when deploying to AWS Amplify. I'm printing out the `NEXTAUTH_URL` in the build process and can see it in the build logs (it's correct). Still redirects to localhost.

**Reply by wallacerenan** [reply]
I found the error, it is not amplifying. Variables without NEXT_PUBLIC at the beginning are not available on the client side, I had to include this value at build time in amplify when uploading the project. on amplify.yml

**Comment by emimaricic** [comment]
Hey, I'm facing the same issue, I know it's been a while but did you guys manage to fix it?

**Comment by nmaties** [comment]
I was able to fix this by adding this line in my build settings - AWS Amplify.
`- echo "NEXTAUTH_URL=$NEXTAUTH_URL" >> .env`

@emimaricic @wallacerenan ^

![image](https://github.com/user-attachments/assets/96a249ba-fff1-4fad-b02f-f988b8a6eca6)

**Reply by aphavichitr** [reply]
> I was able to fix this by adding this line in my build settings - AWS Amplify. `- echo "NEXTAUTH_URL=$NEXTAUTH_URL" >> .env`
> 
> @emimaricic @wallacerenan ^
> 
> ![image](https://private-user-images.githubusercontent.com/16613184/359985573-96a249ba-fff1-4fad-b02f-f988b8a6eca6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1OTE1MjYsIm5iZiI6MTcyODU5MTIyNiwicGF0aCI6Ii8xNjYxMzE4NC8zNTk5ODU1NzMtOTZhMjQ5YmEtZmZmMS00ZmFkLWIwMmYtZjk4OGI4YTZlY2E2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEwVDIwMTM0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4NzZiYmNiMWVhNWU5ZTEyMWUzMWNjOGRhNDY3ZDliZmQxYTk5ODBlMzUwMmE1N2IxNzVmMmZlZjdkNDM4ZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QN2By-ulJtjOw4Zoj-jQxCPeNtxp6JHXvepN20VDYV8)

this worked for me! thanks!
